pyomo/contrib/pynumero/interfaces/external_pyomo_model.py

```diff
@@ -190,14 +283,57 @@ def set_input_values(self, input_values):
         for var, val in zip(input_vars, input_values):
             var.set_value(val)
 
+        vector_scc_idx = 0
         for block, inputs in self._scc_list:
             if len(block.vars) == 1:
                 calculate_variable_from_constraint(
                     block.vars[0], block.cons[0]
                 )
             else:
-                with TemporarySubsystemManager(to_fix=inputs):
-                    solver.solve(block)
+                if self._use_cyipopt:
```

Comment:
`elif` to avoid the extra indentation?

Reply:
See comment below. I think I want to have this extra if statement so that the increment of `vector_scc_idx` happens regardless of whether we're using CyIpopt.